core/src/main/java/org/apache/iceberg/SnapshotManager.java

```diff
@@ -54,6 +54,48 @@ public ManageSnapshots rollbackTo(long snapshotId) {
     return this;
   }
 
+  @Override
+  public ManageSnapshots createBranch(String name, long snapshotId) {
+    transaction.table().manageSnapshots().createBranch(name, snapshotId);
```

Comment:
`manageSnapshots` isn't implemented in transaction tables. I think you need to add package-private operations like `CherrypickOperation` from the last PR.

Reply:
Oh yes definitely. I was mostly going for the API definition here. I can add the operations in this PR as well